test(etdump): Add FileDataSink test coverage

Megan0704-1 opened the pull request:
test(etdump): Add FileDataSink test coverage

## Summary
Adds test validation for FileDataSink usage in ETDumpGen scenarios. Extends existing debug event tests to cover file-based data sinks.

## Test Plan
**Build & Run Tests**:
```bash
# From build directory (cmake-out)
make -j32
ctest -R sdk_etdump_tests --output-on-failure
```

Fixes #9165

Commit:
test(etdump): Add FileDataSink test coverage

Summary:
- Extend test loops to cover FileDataSink scenarios (j=2 case)
- Add temporary file cleanup in TearDown
- Include file_data_sink.h, stdio.h and fstream in test sources
- Update CMakeLists.txt to ensure proper linking

Fixes #9165

Megan0704-1 commented:
@pytorchbot label "topic: not user facing"

Gasoonjia commented on devtools/CMakeLists.txt:
```diff
@@ -233,5 +235,5 @@ install(
 
 if(BUILD_TESTING)
   # TODO: This is currently not working!
```
If you have brought the test building back, plz remove this comment

Gasoonjia commented on devtools/etdump/tests/etdump_test.cpp:
```diff
@@ -45,16 +49,23 @@ class ProfilerETDumpTest : public ::testing::Test {
     const size_t buf_size = 512 * 1024;
     buf = (uint8_t*)malloc(buf_size * sizeof(uint8_t));
     etdump_gen[1] = new ETDumpGen(Span<uint8_t>(buf, buf_size));
+
```
It would be great if you can use the temp file et provided. You can find example here: https://github.com/pytorch/executorch/blob/main/extension/data_loader/test/file_data_loader_test.cpp#L15

Or is there any reason why you didn't choose that?

Megan0704-1 replied:
Thanks for the feedback!!

Gasoonjia commented:
@Megan0704-1 thanks for taking this task! Overall looks great and left some comments. Also some of the cis are broken.

Megan0704-1 committed:
fix(etdump): replace insecure tmpnam with TempFile utility

- Migrate from tmpnam to executorch::extension::testing::TempFile
  for secure temporary file handling
- Add missing namespace qualification for TempFile class
- Use std::unique_ptr for proper RAII management
- Remove manual file deletion since TempFile handles cleanup
- Remove building sdk_etdump_test in devtools/CMakeLists.txt

Gasoonjia commented:
@Megan0704-1 thanks for your contribution! If you like you can join our discord and directly ping us or left message to broader group https://github.com/pytorch/executorch?tab=readme-ov-file#contributing
